### PR TITLE
add Dragonwell on windows

### PR DESCRIPTION
**Pull request by joeyleeeeeee97**
We could use AdoptOpenJDK pipelines to build Dragonwell on windows.

Dragonwell got a building limitation:

Dragonwell 8 Have to depend on bootstrap
So we may need to pull dragonwell windows release during build.

Also, we need this https://github.com/AdoptOpenJDK/openjdk-build/pull/2232 to build dragonwell on windows.

### CODE REVIEW

**Review by karianna — APPROVED** [review]
LGTM - again we need to really split out vendor vs type of VM - maybe we bite the bullet on that refactors in Dec.

**Comment by M-Davies** [comment]
run tests

**Comment by M-Davies** [comment]
> # 🟠 PR TESTER RESULT 🟠
> ❎ Some pipelines failed or the job was aborted! ❎
> _See the `pipeline-build-check` below for more information..._

Existing unrelated problems with sign jobs. Running again since the fix has been merged...

run tests

**Comment by M-Davies** [comment]
> # 🟠 PR TESTER RESULT 🟠
> ❎ Some pipelines failed or the job was aborted! ❎
> _See the `pipeline-build-check` below for more information..._

Unrelated issues

**Review by M-Davies — APPROVED** [review]

**Comment by karianna** [comment]
Merge conflicts

**Comment by sxa** [comment]
Hmm hadn't realised it had to bootstrap with a Dragonwell 8 - I hit an issue with that when trying to build on Linux with GCC9 for one of the other issues (We don't seem to make it available in the non-docker build case on Linux/x64 at the moment) :-)

**Review by sxa — COMMENTED** [review]
Hmm hadn't realised it had to bootstrap with a Dragonwell 8 - I hit an issue with that when trying to build on Linux with GCC9 for one of the other issues (We don't seem to make it available in the non-docker build case on Linux/x64 at the moment) :-)

Why was `checkAndDownloadDragonwellBootstrapWindowsImage` put into `prepareWorkspace.sh` instead of having it added to the ansible scripts on each machine (so as not to download it each time, which is what we do with the other boot JDKs?

**Comment by joeyleeeeeee97** [comment]
> Hmm hadn't realised it had to bootstrap with a Dragonwell 8 - I hit an issue with that when trying to build on Linux with GCC9 for one of the other issues (We don't seem to make it available in the non-docker build case on Linux/x64 at the moment) :-)
> 
> Why was `checkAndDownloadDragonwellBootstrapWindowsImage` put into `prepareWorkspace.sh` instead of having it added to the ansible scripts on each machine (so as not to download it each time, which is what we do with the other boot JDKs?

Yes. If it's the same image we do not need to download it over and over again. We could check if it is already downloaded.
If we download the image into the machine, I wonder where could I find the scripts for initiating machines?

**Comment by joeyleeeeeee97** [comment]
@sxa  https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1727 I added these to ansible but I don't know how to trigger "playbook changes run through VPC or QPC (if you have access)" for tests. 
Could you give the current code another look and help me trigger the tests?

**Comment by joeyleeeeeee97** [comment]
Maybe we could hold this before https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1749, since the downloaded URL is hardcoded may affect this patch.

**Review by sxa — CHANGES_REQUESTED** [review]
I'd rather cover all the settings of the bootstrap within `windows.sh` if possible instead of doing it via `build.sh`. Also can we rename `DRAGONWELL_BOOTSTRAP` to `DRAGONWELL8_BOOTSTRAP` to make it clear to anyone reading the script that it is only used for JDK8 and not all dragonwell versions (It's clear when you look at this PR, but probably not for someone looking just at windows.sh in the future)

**Comment by sxa** [comment]
@joeyleeeeeee97 Just a quick note to say that after the playbook change is merged we'll need to ensure that the playbook updates are deployed on all build systems (Something my team will do!) before merging and running this change.

**Comment by joeyleeeeeee97** [comment]
> @joeyleeeeeee97 Just a quick note to say that after the playbook change is merged we'll need to ensure that the playbook updates are deployed on all build systems (Something my team will do!) before merging and running this change.

I moved all related changes to `windows.sh`.

http://ci.dragonwell-jdk.io/job/build-scripts/job/jobs/job/jdk8u/job/jdk8u-windows-x64-dragonwell/29/console
And I tried a build with an image installed at '/cygdrive/C/openjdk/dragonwell-bootstrap/jdk8u272-ga' and that seems to work. Please check again with https://github.com/AdoptOpenJDK/openjdk-infrastructure/pull/1749. Thanks for your help

**Comment by karianna** [comment]
OK since the requested changes have been made I'll merge this in @sxa's absence.

**Pull request merged**
